packages/core/src/scanning.ts

```diff
@@ -73,25 +73,15 @@ const processTweak = (
     return 0; // No counter increment
 };
 
-export const scanOutputs = (
-    scanPrivateKey: Buffer,
+function performScanLoop(
+    ecdhSecret: Uint8Array,
     spendPublicKey: Buffer,
-    sumOfInputPublicKeys: Buffer,
-    inputHash: Buffer,
     outputs: Buffer[],
+    matches: Map<string, Buffer>,
```

Comment:
instead of accepting this as a parameter, we should just return this from this function